src/services/iota.js

```diff
@@ -72,69 +59,40 @@ const skinnyQueryTransactions = (iotaProvider, addresses) =>
     );
   });
 
-const initializePolling = addresses => {
-  totalLength = addresses.length;
-
-  indexes.startingIdx = 0;
-  indexes.endingIdx = addresses.length - 1;
-  indexes.latestFoundBackIdx = indexes.endingIdx;
-
-  indexes.frontIdx =
-    indexes.startingIdx + Math.floor(Math.random() * BUNDLE_SIZE);
-  indexes.backIdx = indexes.endingIdx - Math.floor(Math.random() * BUNDLE_SIZE);
-};
-
-const checkUploadPercentage = addresses => {
+const checkUploadPercentage = (addresses, frontIndex, backIndex) => {
   let backOfFile = new Promise((resolve, reject) => {
-    skinnyQueryTransactions(IotaA, [addresses[indexes.backIdx]]).then(
+    skinnyQueryTransactions(IotaA, [addresses[backIndex]]).then(
       transactions => {
-        if (transactions.length > 0) {
-          indexes.latestFoundBackIdx = indexes.backIdx;
-          updateBackIndex(indexes.backIdx);
-        }
-        resolve();
+        resolve({
+          backIndex,
+          updateIndex: transactions.length > 0
+        });
       }
     );
   });
 
   let frontOfFile = new Promise((resolve, reject) => {
-    skinnyQueryTransactions(IotaA, [addresses[indexes.frontIdx]]).then(
+    skinnyQueryTransactions(IotaA, [addresses[frontIndex]]).then(
       transactions => {
-        if (transactions.length > 0) {
-          indexes.latestFoundFrontIdx = indexes.frontIdx;
-          updateFrontIndex(indexes.frontIdx);
-        }
-        resolve();
+        resolve({
+          frontIndex,
+          updateIndex: transactions.length > 0
+        });
       }
     );
   });
 
-  return Promise.all([frontOfFile, backOfFile]).then(() => {
-    return recalculatePercentage(indexes);
+  return Promise.all([frontOfFile, backOfFile]).then(indexResults => {
+    return {
+      frontIndex: indexResults[0].frontIndex,
+      updateFrontIndex: indexResults[0].updateIndex,
+      backIndex: indexResults[1].backIndex,
+      updateBackIndex: indexResults[1].updateIndex
```

Comment:
you could do `const [a, b] = indexResults;` (name it something other than a and b of course) and then just do `a. frontIndex` etc. no biggie though just letting you know